### PR TITLE
Doc: Update Supported column for 1.10.x series

### PR DESCRIPTION
**Pull request by kaxil**
Doc: Update Supported column for 1.10.x series

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: docs/apache-airflow/installation/supported-versions.rst [hunk 1]
```diff
@@ -28,7 +28,7 @@ Apache Airflow version life cycle:
 +---------+-----------------+---------------+-----------------+----------------+
 | 2       | Supported       | Dec 17, 2020  | Dec 2021        | TBD            |
```
Comment (eladkal) [inline docs/apache-airflow/installation/supported-versions.rst]: worth also updating this line. Airflow 2 is not limited support for Dec 2021
It's TBD

**Review by potiuk — APPROVED** [review]
How come we have not updated it yet :)

File: docs/apache-airflow/installation/supported-versions.rst (hunk 1 above)
Comment (potiuk) [reply docs/apache-airflow/installation/supported-versions.rst]: Ah yeah

Comment (potiuk) [reply docs/apache-airflow/installation/supported-versions.rst]: Hopefully :D

Comment (kaxil) [reply docs/apache-airflow/installation/supported-versions.rst]: done -https://github.com/apache/airflow/pull/20592/commits/3900e1224037d9691e0fcc82b97245775fe1d2a6

**Comment by eladkal** [comment]
@potiuk because for some reason we maintain this table twice :( 
Readme:
https://github.com/apache/airflow#version-life-cycle
which is updated!
And apparently also in rst

**Review by eladkal — APPROVED** [review]

**Comment by potiuk** [comment]
Hold my beer!

**Comment by potiuk** [comment]
https://github.com/apache/airflow/pull/20594

**Pull request merged**